packages/flutter_tools/lib/src/runner/flutter_command_runner.dart

```diff
@@ -183,6 +183,12 @@ class FlutterCommandRunner extends CommandRunner<void> {
     try {
       if (platform.script.scheme == 'data')
         return '../..'; // we're running as a test
+
+      if (platform.script.scheme == 'package') {
+        String packageConfigPath = Uri.parse(platform.packageConfig).toFilePath();
+        return fs.path.dirname(fs.path.dirname(fs.path.dirname(packageConfigPath)));
```

Comment:
Perhaps add a comment regarding directory structure we expect? I'm not sure what this navigation three levels up mean? Given 'package:flutter_gallery/main.dart' we go three levels up from .packages file, is that right?

Reply:
The paths in `_defaultFlutterRoot` are based on the directory structure of `packages/flutter_tools` in the Flutter framework repository.  The packageConfig for `flutter_tools` lives at `packages/flutter_tools/.packages`, so the Flutter root is three levels up from there.